test/cppsuite/test_harness/workload_generator.h

```diff
@@ -138,13 +130,22 @@ class workload_generator : public component {
     void
     finish()
     {
+        int error_code = 0;
+
         debug_info("Workload generator stage done", _trace_level, DEBUG_INFO);
         for (const auto &it : _workers) {
             it->finish();
         }
         _thread_manager.join();
     }
 
+    void
+    set_tracker(workload_tracking *tracking)
+    {
+        _enable_tracking = (tracking != nullptr);
```

Comment:
This is kind weird, I think calling set tracker would imply that tracking isn't null and this statement should just be, might be worth putting an error check to make sure people are using it correctly:
```
testutil_check(tracking != nullptr);
_enable_tracking = true
```